src/Controller/MediaController.php

```diff
@@ -42,51 +44,49 @@ public function getMedia($id)
     }
 
     /**
-     * @throws NotFoundHttpException
      *
+     * @param Request $request
      * @param string $id
      * @param string $format
-     *
      * @return Response
      */
-    public function downloadAction($id, $format = MediaProviderInterface::FORMAT_REFERENCE)
+    public function downloadAction(Request $request, $id, $format = MediaProviderInterface::FORMAT_REFERENCE)
```

Comment:
This is a BC-break, please use something like this: https://github.com/sonata-project/SonataAdminBundle/blob/3.x/src/Controller/CRUDController.php#L961

Reply:
@greg0ire reverted the request call back to original.